corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowserEditor.java

```diff
@@ -188,15 +194,45 @@ private void printPayload(Map.Entry<CorfuDynamicKey, CorfuDynamicRecord> entry)
             log.error("payload is NULL");
             return;
         }
+
         try {
             builder = new StringBuilder("\nPayload:\n")
                     .append(JsonFormat.printer().print(entry.getValue().getPayload()));
-            log.info(builder.toString());
+            System.out.println(builder.toString());
         } catch (Exception e) {
             log.error("invalid payload: ", e);
         }
     }
 
+    private int printTableRegistry() {
+        for (Map.Entry<TableName, CorfuRecord<TableDescriptors, TableMetadata>> entry :
+                dynamicProtobufSerializer.getCachedRegistryTable().entrySet()) {
+            try {
+                StringBuilder builder = new StringBuilder("\nKey:\n")
+                        .append(JsonFormat.printer().print(entry.getKey()));
+                System.out.println(builder.toString());
+            } catch (Exception e) {
+                log.error("Unable to print metadata section of registry table");
+            }
+            System.out.println("\nkeyType = \""+entry.getValue().getPayload().getKey().getTypeUrl()+"\"");
+            System.out.println("\npayloadType = \""+entry.getValue().getPayload().getValue().getTypeUrl()+"\"");
+            System.out.println("\nmetadataType = \""+entry.getValue().getPayload().getMetadata().getTypeUrl()+"\"");
+            System.out.println("\nProtobuf Source Files: {\""+
+                    entry.getValue().getPayload().getFileDescriptorsMap().keySet()
+            );
+
+            try {
+                StringBuilder builder = new StringBuilder("\nMetadata:\n")
+                        .append(JsonFormat.printer().print(entry.getValue().getMetadata()));
```

Comment:
Can we use the printMetadata() method instead?

Reply:
No printMetadata() works with CorfuDynamicRecord types while this section deals directly with the primitive type, so the types dont match.